recipes-extended/xen/files/0001-convert-block-tap-for-shared.patch

```diff
@@ -0,0 +1,551 @@
+From 164436bdff5620e605268ec82860728014dab7cb Mon Sep 17 00:00:00 2001
```

Comment:
The commit message introducing it has:
    xen: Patch block-tap script
    
    This brings in a patched block-tap script.  It provides the device
    sharing checks previously done in libxl.
    
    Calling tap-ctl passes an open file descriptor to tapback which messes
    up locking.  That is why we have `eval "exec $_lockfd>&-` to close the
    flock fd.
    
    We also play games for the keydir.  This tries to mimic what's in xenmgr
    already... I guess we could let xenmgr set those variables before
    execing xl...

I can expand the patch description with something like:
Make block-tap usable with OpenXT

Xen has a block-tap script to handle vhds, but it needs some changes to be usable by OpenXT.

It needs to handle device sharing checks which were previously done in libxl.  This is partially accomplished by moving and reusing functions from Xen's block script to common code.

The call to tap-ctl needs to change to prevent leaking the open _lockfd file descriptor into the tapdisk daemon.  The `eval "exec $_lockfd>&-` call closes the fd to prevent that leak.  Such a leak blocks any other block-tap script from running thereby blocking any other VM from starting.
 
It relies on TAPDISK3_CRYPTO_KEYDIR being set in the environment by xenmgr to allow tapdisk to unlock encrypted VHDs.

Reply:
Appreciate the explanation! Since you already wrote it you might as well add it.